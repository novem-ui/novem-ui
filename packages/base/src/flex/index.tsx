import { css } from '@emotion/react'

import { FlexProps } from '../types'

const flex = ({ direction, wrap, basis, grow, shrink, align, justify }: FlexProps) => css`
  ${align && `align-items: ${align};`}
  ${basis && `flex-basis: ${basis};`}
  ${justify && `justify-content: ${justify};`}
  ${direction && `flex-direction: ${direction};`}
  ${grow && `flex-grow: ${grow};`}
  ${shrink && `flex-shrink: ${shrink};`}
  ${wrap && `flex-wrap: ${wrap};`}
`

export default flex
