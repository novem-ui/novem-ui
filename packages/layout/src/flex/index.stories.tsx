import React, { PropsWithRef } from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Flex, FlexProps } from '.'

export default {
  title: 'Components/Layout/Flex',
  component: Flex
} as Meta

const Template: Story<PropsWithRef<FlexProps>> = (props) => <Flex {...props}>text content</Flex>

export const Default = Template.bind({})

Default.args = {
  backgroundColor: 'orange.100',
  height: '100px',
  borderRadius: '10px',
  borderWidth: '1px',
  borderStyle: 'solid',
  borderColor: 'orange.700',
  align: 'center',
  justify: 'center'
}
