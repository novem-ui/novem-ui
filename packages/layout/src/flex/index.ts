import styled from '@emotion/styled'
import { flex, FlexProps as FlexboxProps } from '@novem-ui/base'

import { Box, BoxProps } from '../box'

export type FlexProps = Omit<BoxProps, 'display'> & FlexboxProps & { inline?: boolean }

export const Flex = styled(Box)<FlexProps>`
  display: ${({ inline }) => (inline ? 'inline-flex' : 'flex')};
  ${flex}
`
