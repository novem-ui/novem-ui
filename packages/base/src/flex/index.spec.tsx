import React from 'react'

import { render } from '@testing-library/react'
import styled from '@emotion/styled'
import { matchers } from '@emotion/jest'

import flex from '.'

import type { FlexProps } from '../types'

beforeAll(() => {
  expect.extend(matchers)
})

const FlexComponent = styled.div<FlexProps>`
  ${flex}
`

describe('borderColor', () => {
  test('should return a component some flex props', () => {
    const { container } = render(<FlexComponent align="center" />)

    expect(container).toMatchSnapshot()
  })

  test('should return a component all the flex props', () => {
    const props: FlexProps = {
      align: 'center',
      justify: 'center',
      wrap: 'wrap',
      grow: 1,
      basis: 'max-content',
      direction: 'row',
      shrink: 'inherit'
    }
    const { container } = render(<FlexComponent {...props} />)

    expect(container).toMatchSnapshot()
  })
})
