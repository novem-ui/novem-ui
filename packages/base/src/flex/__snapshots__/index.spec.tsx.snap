// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`borderColor should return a component all the flex props 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: max-content;
  -ms-flex-preferred-size: max-content;
  flex-basis: max-content;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: inherit;
  -ms-flex-negative: inherit;
  flex-shrink: inherit;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div
    class="emotion-0 emotion-1"
    direction="row"
    wrap="wrap"
  />
</div>
`;

exports[`borderColor should return a component some flex props 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;
